frontend/src/queries/nodes/InsightViz/InsightViz.tsx

```diff
@@ -36,10 +36,11 @@ let uniqueNode = 0
 
 export function InsightViz({ uniqueKey, query, setQuery, context, readOnly }: InsightVizProps): JSX.Element {
     const [key] = useState(() => `InsightViz.${uniqueKey || uniqueNode++}`)
-    const insightProps: InsightLogicProps = context?.insightProps || {
+    const insightProps: InsightLogicProps = {
         dashboardItemId: `new-AdHoc.${key}`,
         query,
         setQuery,
+        ...(context?.insightProps || {}),
```

Comment:
Flagging this change - I don't have enough context to be 100% certain this is safe

Reply:
I'm not sure either... I see the main change is that you added three things to this `insightLogicProps`:

```ts
    hidePersonsModal?: boolean
    suppressSessionAnalysisWarning?: boolean
    chartRenderingMetadata?: ChartRenderingMetadata
```

The existing insight logic props have to do with just the insight ID or cached results, these are very specific toggles for some specific insights. My first reaction is that they feel out of place.  Since they do affect how the insight is displayed, should/could they just be part of the query node itself?

Reply:
Do you think they make more sense as part of `QueryContext`?

Reply:
Hmm I guess putting them on the InsightVizNode could also work

Reply:
Toggling these will not affect the fetched data, so the InsightVizNode seems like the better option indeed.

Reply:
Ah the `chartRenderingMetadata` contains a function so can't go inside the query node

Reply:
Could that be passed in the context for the `<Query>` tag?

Reply:
Yeah, we currently don't pass this into the Trends component, but some of the ActionX components can receive it as a prop, so I can change how this is wired up.

My plan is to split this, with the chart rendering functions moving to the context, and the other ones moving to the InsightViz node.

Sent from my iPhone

Reply:
This re-wiring is done, I only touched the TrendInsight component and not the others e.g. FunnelInsight but it would be very simple to wire those up the same way if needed later